server/api/src/rsGeneralAdmin.cpp

```diff
@@ -1135,9 +1139,43 @@ _rsGeneralAdmin( rsComm_t *rsComm, generalAdminInp_t *generalAdminInp ) {
     }
     if ( strcmp( generalAdminInp->arg0, "rm" ) == 0 ) {
         if ( strcmp( generalAdminInp->arg1, "user" ) == 0 ) {
-            return irods::remove_user(*rsComm,
-                                      generalAdminInp->arg2 ? generalAdminInp->arg2 : "",
-                                      generalAdminInp->arg3 ? generalAdminInp->arg3 : "");
+            const auto* user_name = generalAdminInp->arg2 ? generalAdminInp->arg2 : "";
+            const auto* zone_name = generalAdminInp->arg3 ? generalAdminInp->arg3 : "";
+            namespace iua = irods::experimental::administration;
+            const auto user_and_zone = irods::user::validate_name(user_name);
```

Comment:
What happens if `user_name` is invalid?
Does it throw or return an empty optional?
Should we care?

Reply:
That sounds correct.

The call to `validate_name` should only happen if we're dealing with a 4.3.2 server. You agree?

Reply:
Good point. I'll add that.